src/NuGet.Clients/NuGet.PackageManagement.VisualStudio/Projects/LegacyPackageReferenceProject.cs

```diff
@@ -376,8 +376,11 @@ private async Task<PackageSpec> GetPackageSpecAsync(ISettings settings)
             AssetTargetFallbackUtility.ApplyFramework(projectTfi, packageTargetFallback, assetTargetFallback);
 
             // Build up runtime information.
-            var runtimes = await _vsProjectAdapter.GetRuntimeIdentifiersAsync();
-            var supports = await _vsProjectAdapter.GetRuntimeSupportsAsync();
+
+            var runtimes = GetRuntimeIdentifiers(
+                GetPropertySafe(_vsProjectAdapter.BuildProperties, ProjectBuildProperties.RuntimeIdentifier),
+                GetPropertySafe(_vsProjectAdapter.BuildProperties, ProjectBuildProperties.RuntimeIdentifiers));
+            var supports = GetRuntimeSupports(GetPropertySafe(_vsProjectAdapter.BuildProperties, ProjectBuildProperties.RuntimeSupports));
```

Comment:
I'm not a fan of nesting method calls. When debugging, if I want to step into GetRuntimeIdentifiers, I don't want to step into two calls of GetPropertySafe that I then need to step out of. Or, in order to see the values passed into GetRuntimeIdentifiers, I have to step into it, because they're not local variables before the method is called.

Reply:
I see that, and I get it. I run into that sometimes myself, but having the assignment would add a lot of long to methods that are already pretty long and imo mess with readability even further.